ga4gh/avrotools.py

```diff
@@ -129,309 +92,419 @@ def _ensureProtocolSubclass(self):
-        generator from the supplied list.
-        """
-        while True:
-            yield random.choice(generators).next()
+    def handleFloat(self, min_=INT_MIN_VALUE, max_=INT_MAX_VALUE):
+        return random.uniform(min_, max_)
 
-    @classmethod
-    def _arrayGenerator(cls, generator, length=10):
-        """
-        Creates a generator that returns an an array of values taken from the
-        supplied generator.
-        """
-        while True:
-            yield list(itertools.islice(generator, length))
+    def handleDouble(self, min_=LONG_MIN_VALUE, max_=LONG_MAX_VALUE):
+        return random.uniform(min_, max_)
 
-    @classmethod
-    def _mapGenerator(cls, valueGen, size=10):
-        """
-        Creates a generator that returns an a dictionary with keys generated
-        by keyGen and values generated by valueGen.
-        """
-        keyGen = cls._stringGenerator()
-        while True:
-            keys = list(itertools.islice(keyGen, size))
-            values = list(itertools.islice(valueGen, size))
-            yield dict(zip(keys, values))
-
-    @classmethod
-    def _recordGenerator(self, recordSchema):
-        """
-        Creates a generator which returns a dictionary where keys and values
-        are determined by the recordSchema. The recordSchema is a map of the
-        records keys to a generator of valid values for that key.
-        """
-        while True:
-            record = {}
-            for key, gen in recordSchema.iteritems():
-                record[key] = gen.next()
-            yield record
+    def handleFixed(self, schema):
+        return self.handleString(schema.size)
 
-    def __init__(self, class_):
-        self.class_ = class_
-        self.schema = class_.schema
+    def handleEnum(self, schema):
+        return random.choice(schema.symbols)
+
+    def handleArray(self, schema, length=10):
+        toReturn = []
+        for _ in xrange(length):
+            item = self.handleSchema(schema.items)
+            toReturn.append(item)
+        return toReturn
+
+    def handleMap(self, schema, size=10):
+        toReturn = {}
+        for _ in xrange(size):
+            key = self.handleString()
+            value = self.handleSchema(schema.values)
+            toReturn[key] = value
+        return toReturn
+
+    def handleUnion(self, schema):
+        chosenSchema = random.choice(schema.schemas)
+        return self.handleSchema(chosenSchema)
 
+    def handleRecord(self, schema):
+        record = {}
+        for field in schema.fields:
+            record[field.name] = self.handleSchema(field.type)
+        return record
+
+
+class TypicalInstanceCreator(AvroTypeSwitch):
+    """
+    Generates typical instances and values
+    """
+    def handleNull(self):
+        raise SchemaToolException()
+
+    def handleBoolean(self):
+        return True
+
+    def handleString(self):
+        return 'aString'
+
+    def handleBytes(self):
+        return 'someBytes'
```

Comment:
Should have a 'b' prefix here for Python 3 compatibility:
```python
return b'someBytes'
```

I know we're not supporting Python 3 right now, but I think it's worth doing these little things to make the job easier when we do change.